applications/Chat/tests/test_dataset.py

```diff
@@ -213,6 +214,20 @@ def test_sft_dataset(model: str,
                                             max_length=max_length)
         assert len(sft_dataset) == min(max_dataset_size, len(SFT_DATASET))
 
+    if isinstance(tokenizer, ChatGLMTokenizer):
+        for i in range(max_dataset_size):
+            assert isinstance(sft_dataset[i], dict)
+            assert list(sft_dataset[i].keys()) == ["input_ids", "labels", "attention_mask"]
+            input_ids = sft_dataset[i]["input_ids"]
+            labels = sft_dataset[i]["labels"]
+            attention_mask = sft_dataset[i]["attention_mask"].to(torch.bool)
+            assert input_ids.shape == labels.shape == torch.Size([max_length])
+            assert attention_mask.shape[1] == attention_mask.shape[2] == max_length
+            
+            ignore_mask = labels == IGNORE_INDEX
+            assert input_ids.masked_select(torch.logical_not(ignore_mask))[0] == tokenizer.bos_token_id
+            return
+        
```

Comment:
As you have implemented a special pre-processing method for chatglm. Your tests should test if the output follows the correct process and generate the output correctly.

Reply:
Added